parallel-install/pkg/engine/engine.go

```diff
@@ -116,6 +141,11 @@ func run(ctx context.Context, statusChan chan<- components.Component, cmps []com
 	wg.Wait()
 }
 
+//Non-blocking worker.
+//Designed to run in parallel (several workers are processing the same jobChan).
+//Detects Context cancellation.
+//Context cancellation is not detected immediately, but between component processing operations - because such operations are blocking.
```

Comment:
```suggestion
//Context cancellation is not detected immediately. It's detected between component processing operations because such operations are blocking.
```